src/Val.cc

```diff
@@ -3202,7 +3202,13 @@ ValPtr EnumVal::DoClone(CloneState* state) {
     return {NewRef{}, this};
 }
 
-void TypeVal::ValDescribe(ODesc* d) const { d->Add(type->AsTypeType()->GetType()->GetName()); }
+void TypeVal::ValDescribe(ODesc* d) const {
+    const auto& ty = type->AsTypeType()->GetType();
+    if ( ty->GetName().empty() )
+        d->Add(type_name(ty->Tag()));
+    else
+        d->Add(ty->GetName());
+}
```

Comment:
WDYT about using `type->Describe(ODesc *)` instead?

Reply:
Yeah I'll change to that. I figured for something tiny changing less would be preferred and I didn't want to change existing type vals, but turns out with a name a lot will just print the name anyway... so that's a wash :)